chromium_src/chrome/renderer/worker_content_settings_client.cc

```diff
@@ -46,4 +48,36 @@ bool WorkerContentSettingsClient::AllowFingerprinting(
   return GetBraveFarblingLevel() != BraveFarblingLevel::MAXIMUM;
 }
 
+blink::WebSecurityOrigin
+WorkerContentSettingsClient::GetEphemeralStorageOriginSync() {
+  if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
+    return {};
+
+  if (is_unique_origin_)
+    return {};
+
+  // If first party ephemeral storage is enabled, we should always ask the
+  // browser if a worker should use ephemeral storage or not.
+  if (!base::FeatureList::IsEnabled(
+          net::features::kBraveFirstPartyEphemeralStorage) &&
+      net::registry_controlled_domains::SameDomainOrHost(
+          top_frame_origin_, document_origin_,
+          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
+    return {};
+  }
+
+  EnsureContentSettingsManager();
+
+  absl::optional<url::Origin> optional_ephemeral_storage_origin;
+  content_settings_manager_->AllowEphemeralStorageAccess(
+      render_frame_id_, document_origin_, site_for_cookies_, top_frame_origin_,
+      &optional_ephemeral_storage_origin);
+  // Don't cache the value intentionally as other WorkerContentSettingsClient
+  // methods do.
+  return blink::WebSecurityOrigin(
+      optional_ephemeral_storage_origin
+          ? blink::WebSecurityOrigin(*optional_ephemeral_storage_origin)
+          : blink::WebSecurityOrigin());
```

Comment:
doesn't this create an opaque origin? Shouldn't we return null here?

Reply:
so is that different from https://github.com/brave/brave-core/pull/12686/files#diff-430f56739d7df3701057dcb1c3ea5412094160cc91008cfd596d42d889e4d8d4R66 ? It's a bit confusing that we're returning seemingly different things there

Reply:
yep, but just for easier reading maybe it's better to always return `blink::WebSecurityOrigin()`?

Reply:
I think it's fine. Same approach is used in many places.